packages/squiggle-lang/src/dist/SymbolicDist.ts

```diff
@@ -854,6 +855,103 @@ export class Bernoulli extends SymbolicDist {
     );
   }
 }
+export class Metalog extends SymbolicDist {
+  a: number[];
+  private constructor({ a }: { a: number[] }) {
+    super();
+    this.a = a;
+  }
+
+  static make({
+    a,
+  }: {
+    a: number[];
+  }): result<Metalog, string> {
+    const validationResult = metalog.validate(a);
+    if(validationResult === metalog.MetalogValidationStatus.Success){
+      return Ok(new Metalog({a}))
+    }
+    else if(validationResult === metalog.MetalogValidationStatus.NotEnoughParamaters){
+      return Result.Error("Terms array must have more than one element");
+    }
+    else {
+      return Result.Error("Invalid combination of a (pdf is negative in parts)");
```

Comment:
This relies on a secret knowledge that there are just three error types in `metalog.validate`; that might change in the future.

So it'd be safer to check for `MetalogValidationStatus.NegativeDerivative` explicitly and fail with "unknown validation error" in the `else` branch.